### ISSUE TITLE
Allow enabling multiple texture packs with precedency

### ISSUE THREAD

**Issue by appgurueu** [minetest/minetest#11786]
See https://github.com/minetest/minetest/issues/10829#issuecomment-781793942

**Comment by AFCMS** [comment]
100% aggreed

**Comment by jkf16m** [comment]
mmm hi, let me see if I understand @appgurueu 

This feature request is about, being able to select more than one texture pack (because right now, we can only have one enable per time).

So, in a new menu for example, a new tab, we could select which textures pack to have enable, and select the order in which one will precede to the other, in case a certain texture was already found.

For example, let's say we have the texture packs: "default", "TX1" and "TX2"

having default as the first one, will be overridable for any texture pack (being the top of the list).
then TX1 will be the second one, overriding the textures "door" and "grass"
and then TX2 will be the third one overriding the "door" texture.


If this is what I understood, actually the same design minecraft follows.

**Comment by jkf16m** [comment]
the only problem that I see is, how should we organize "packages"? from what I've seen, packages tab includes:
- texture packs
- mods
- games

![image](https://user-images.githubusercontent.com/39142605/182207377-cead5522-4286-41b5-afe8-37fcc0a6803a.png)

shouldn't we make subtabs of some sort, inside packages? or we can make this feature work without the need of making new tabs?.

![image](https://user-images.githubusercontent.com/39142605/182207591-20cde940-163a-4524-9eae-72e510a478f5.png)

**Comment by NA0341** [comment]
I agree with @jkf16m: we should make subtabs. And we should also consider the fact that there might be more than textures in the future. So essentially we're talking about resources in general.

I suggest three tabs:

+ Games
+ Mods
+ Resources

In the Resources Tab, you can (de)activate & reorder packs. With the Information Section to the right remaining as is.

+ This is a very simple approach that doesn't conflict with the way Minetest's UI is built.
+ Splitting Content into Tabs brings the benefit of enhanced UX - especially when dealing with lots of game resources.
+ The ability to arrange packages regardless of whether they are enabled or disabled enables flexible use of a wide variety of scenarios. In other words: You can have arrangements for several scenarios and easily switch between them by enabling or disabling the respective packs.
+ The KISS approach I described also offers better UX than in MC's implementation!

This should do the trick for the most common scenarios.


**I want to highlight again that this is not just some convenience thing. Putting together a new pack from all resources you want to use has Major Drawbacks!**

+ You loose automatic texture updates.
+ If you want something else, you may need to start over from scratch.
+ You need to upload your new pack in order to share your configuration.
+ etc.

**Comment by rollerozxa** [comment]
The absolute minimum to support enabling multiple texture packs on the client-side (excluding a graphical dialog for enabling and moving texpacks up and down) would be to make `getTextureDirs()` be able to split `texture_path` on a separator and return a list of paths:
```c++
std::vector<std::string> getTextureDirs()
{
	std::vector<std::string> paths;

	std::vector<std::string> texture_paths = str_split(g_settings->get("texture_path"), ':');
	for (auto path : texture_paths)
		fs::GetRecursiveDirs(paths, path);

	return paths;
}
```
This will take care of priority, i.e. the firstmost path is top priority and then it goes downwards when looking for textures.

**Comment by ZenonSeth** [comment]
How's this look?

<img width="1460" height="917" alt="Image" src="https://github.com/user-attachments/assets/6e5bf63d-b7be-4ee0-9c60-8bc5b39cbaf5" />
